src/webapp/components/survey/TableSection.tsx

```diff
@@ -0,0 +1,136 @@
+import { Typography } from "@material-ui/core";
+import { Question } from "../../../domain/entities/Questionnaire/QuestionnaireQuestion";
+import { QuestionnaireSection } from "../../../domain/entities/Questionnaire/QuestionnaireSection";
+import { QuestionWidget } from "../survey-questions/QuestionWidget";
+import { PaddedDiv, StyledInput, StyledTitle, StyledWrapper } from "./SurveySection";
+import {
+    // @ts-ignore
+    DataTable,
+    // @ts-ignore
+    TableHead,
+    // @ts-ignore
+    DataTableRow,
+    // @ts-ignore
+    DataTableColumnHeader,
+    // @ts-ignore
+    DataTableCell,
+    // @ts-ignore
+    TableBody,
+} from "@dhis2/ui";
+import _c from "../../../domain/entities/generic/Collection";
+import styled from "styled-components";
+import { useTableSection } from "./hook/useTableSection";
+
+interface TableSectionProps {
+    section: QuestionnaireSection;
+    viewOnly?: boolean;
+    updateQuestion: (question: Question) => void;
+}
+
+export const TableSection: React.FC<TableSectionProps> = ({
+    section,
+    viewOnly,
+    updateQuestion,
+}) => {
+    const { questionGroups, headerRow } = useTableSection(section);
+    return (
+        <div key={section.title}>
+            <DataTable>
+                <TableHead>
+                    <DataTableRow>
+                        <DataTableColumnHeader colSpan="6">
+                            <StyledTitle>{section.title}</StyledTitle>
+                        </DataTableColumnHeader>
+                    </DataTableRow>
+                    <DataTableRow>
+                        {headerRow?.map((header, index) => (
+                            <DataTableColumnHeader key={index}>{header}</DataTableColumnHeader>
+                        ))}
+                    </DataTableRow>
+                </TableHead>
+
+                <TableBody>
+                    {questionGroups?.map(questionGroup => (
+                        <>
+                            <DataTableRow key={`columns-${questionGroup.groupId}`}>
+                                {questionGroup.columnQuestions.map(
+                                    question =>
+                                        question.isVisible && (
+                                            <DataTableCell key={question.code}>
```

Comment:
(code-only review) This part of the code is very similar to the code below, and also to the code in SurveySection.tsx. If you have time, maybe would be interesting to merge and create just one component to be use in TableSection and in SurveySection components